05-onchain.md

```diff
@@ -523,67 +523,70 @@ irrevocably resolved, should still protect against this happening. [ FIXME: May
+non-witness data: assuming it has a pay-to-witness-script-hash (the largest
+standard output script), in addition to a 2-byte witness header.
 
-In addition to outputs being swept under as penalty, the node MAY also sweep the
+In addition to the outputs being swept under [ FIXME: what does 'swept under' mean? ] as penalty, the node MAY also sweep the
 `to_remote` output of the commitment transaction (e.g. to reduce the total
-amount paid in fees). Doing so requires the inclusion of a p2wpkh witness and
-additional txinput, resulting in an additional 108 + 164 = 272 bytes.
+amount paid in fees). Doing so requires the inclusion of a P2WPKH witness and an
+additional *txinput*, resulting in an additional 108 + 164 = 272 bytes.
 
-In a worst case scenario, a node holds only incoming HTLCs, and the HTLC-timeout
-transactions are not published, which forces the node to spend from the commitment
-transaction.
+In the worst case scenario, the node holds only incoming HTLCs, and the
+HTLC-timeout transactions are not published, which forces the node to spend from
+the commitment transaction.
 
-With a maximum standard weight of 400000, the maximum number of HTLCs that can
-be swept in a single transaction:
+With a maximum standard weight of 400000 bytes, the maximum number of HTLCs that
+can be swept in a single transaction is as follows:
 
     max_num_htlcs = (400000 - 324 - 272 - (4 * 53) - 2) / 413 = 966
 
-This allows 483 HTLCs in each direction (with both `to_local` and
-`to_remote` outputs) to still be resolved in a single penalty transaction.
-Note that even if the `to_remote` output is not swept, the resulting
-`max_num_htlcs` is 967, which yields the same unidirectional limit of 483 HTLCs.
+Thus, 483 bidirectional (containing both `to_local` and
+`to_remote` outputs) HTLCs can be resolved in a single penalty transaction.
+Note: even if the `to_remote` output is not swept, the resulting
+`max_num_htlcs` is 967; which yields the same unidirectional limit of 483 HTLCs.
 
 # General Requirements
 
-A node SHOULD report an error to the operator if it discovers a transaction
-spend the funding transaction output which does not fall into one of
-these categories (mutual close, unilateral close, or revoked
-transaction close). Such a transaction implies its private key has
-leaked, and funds may be lost.
-
-A node MAY simply watch the contents of the most-work chain for
-transactions, or MAY watch for (valid) broadcast transactions a.k.a
-mempool. Considering mempool transactions should cause lower latency
-for HTLC redemption, but on-chain HTLCs should be such an unusual case
-that speed cannot be considered critical.
+A node:
+  - upon discovering a transaction that spends a funding transaction output
+  which does not fall into one of the above categories (mutual close, unilateral
+  close, or revoked transaction close):
+    - SHOULD report an error to the operator [ FIXME: who is the 'operator'? ].
```

Comment:
Let's change this to the same language as above?  Something like "MUST warn about lost funds" IIRC?

Reply:
changed